File: exercise/day16/src/test/java/blog/ArticleBuilder.java
package blog;

import java.time.LocalDate;
import java.util.HashMap;
import java.util.Map;

public class ArticleBuilder {
    public static final String AUTHOR = "Pablo Escobar";
    public static final String COMMENT_TEXT = "Amazing article !!!";
    private final HashMap<String, String> comments;

    public ArticleBuilder() {
        comments = new HashMap<>();
    }

    public static ArticleBuilder anArticle() {
        return new ArticleBuilder();
    }

    public ArticleBuilder commented() {
        this.comments.put(COMMENT_TEXT, AUTHOR);
        return this;
    }
    public ArticleBuilder commented(String text, String author) {
        this.comments.put(text, author);
        return this;
    }

    public Article build() throws CommentAlreadyExistException {
        return new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore",
                comments.entrySet().stream()
                        .map(e -> new Comment(e.getKey(), e.getValue(), LocalDate.now()))
                        .toArray(Comment[]::new)
        );
    }
}

File: exercise/c#/Day19/Day19.Tests/ArticleBuilder.cs
using LanguageExt;

namespace Day19.Tests
{
	using static Either<Error, Article>;

	public class ArticleBuilder
	{
		public const string Author = "Pablo Escobar";
		public const string CommentText = "Amazing article !!!";

		private readonly Dictionary<string, string> _comments = new();
		private readonly Bogus.Randomizer _random = new();

		public static ArticleBuilder AnArticle() => new();

		public ArticleBuilder Commented()
		{
			_comments.Add(CommentText, Author);
			return this;
		}

		public Either<Error, Article> Build()
		{
			return _comments
						.Aggregate(
							Right(new Article(_random.String(), _random.String())),
							(article, comment) => article.Right(a => a.AddComment(comment.Key, comment.Value)).Left(e => e)
						);
		}
	}
}